File: 009_Palindrome_Number/solution.go
const (
    MAX int = 1<<31 - 1
)
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }
    tmp := x
    reverse := 0
    for ; tmp > 0; {
        reverse = 10 * reverse + tmp % 10
        if reverse > MAX {
            return false
        }
        tmp = tmp / 10
    }
    return (reverse == x)
}

File: 009_Palindrome_Number/solution.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        origin = x
        reverse = 0
        while x > 0:
            reverse = reverse * 10 + x % 10
            if reverse >= math.pow(2, 31):
                return False
            x = x / 10
        return (reverse == origin) 
